File: scripts/hooks/console-log-warn.js
#!/usr/bin/env node
/**
 * console-log-warn.js — Warn about console.log in edited files
 *
 * Advisory (exit 0). Reminds to remove debug statements.
 */

const fs = require('fs');

try {
  const raw = fs.readFileSync(0, 'utf8');
  const input = JSON.parse(raw);
  const content = (input.tool_input && (input.tool_input.new_string || input.tool_input.content)) || '';

  if (!content) process.exit(0);

  const debugPatterns = [
    /console\.log\(/,
    /console\.debug\(/,
    /debugger;/,
    /print\(\s*f?["']/,  // Python print
  ];

  const found = debugPatterns.filter(p => p.test(content));
  if (found.length > 0) {
    console.log(
      'SuperOPC: Debug statement detected in edited content. ' +
      'Remember to remove console.log/debugger/print before committing.'
    );
  }
} catch (e) {
  // Silent pass
}

process.exit(0);

File: scripts/hooks/console_log_warn.py
#!/usr/bin/env python3

from __future__ import annotations

import re
import sys

from common import get_first_content, get_tool_input, read_stdin_json, write_message


DEBUG_PATTERNS = [
    re.compile(r"console\.log\("),
    re.compile(r"console\.debug\("),
    re.compile(r"debugger;"),
    re.compile(r"print\(\s*f?[\"']"),
]


def main() -> int:
    payload = read_stdin_json()
    content = get_first_content(get_tool_input(payload))
    if not content:
        return 0

    if any(pattern.search(content) for pattern in DEBUG_PATTERNS):
        write_message(
            "SuperOPC: Debug statement detected in edited content. "
            "Remember to remove console.log/debugger/print before committing."
        )

    return 0


if __name__ == "__main__":
    sys.exit(main())
